crates/bevy_app/src/app.rs

```diff
@@ -825,11 +827,58 @@ impl App {
     where
         T: Plugin,
     {
+        self.add_boxed_plugin(Box::new(plugin))
+    }
+
+    /// Boxed variant of `add_plugin`, can be used from a [`PluginGroup`]
+    pub(crate) fn add_boxed_plugin(&mut self, plugin: Box<dyn Plugin>) -> &mut Self {
         debug!("added plugin: {}", plugin.name());
         plugin.build(self);
+        self.plugin_registry.push(plugin);
         self
     }
 
+    /// Checks if a [`Plugin`] has already been added.
+    ///
+    /// This can be used by plugins to check if a plugin dependency they have has already been
+    /// added.
+    pub fn is_plugin_added<T>(&self) -> bool
+    where
+        T: Plugin,
+    {
+        self.plugin_registry
+            .iter()
+            .find(|p| p.downcast_ref::<T>().is_some())
+            .is_some()
+    }
+
+    /// Returns the plugin of a given type that have been added.
+    ///
+    /// This can be used by plugins to get the settings of already added plugins.
```

Comment:
```suggestion
    /// This can be used to read the settings of any already added plugins.
    /// This vector will be length zero if no plugins of that type have been added.
    /// If multiple copies of the same plugin are added to the [`App`], they will be listed in insertion order in this vector.
```